.github/workflows/check-and-manage-shavar-files-updates.yml

```diff
@@ -0,0 +1,116 @@
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - uses: actions/checkout@v3
+      with:
+        repository: 'irios3/shavar-list-creation'
+        ref: 'creat-json-tracking-protection-lists'
+    - id: 'auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        credentials_json: '${{ secrets.GCLOUD_AUTH }}'
+        export_environment_variables: true
+    - name: Download version file
+      run: |
+        echo "Download version.txt file"
+        curl -X GET -o "version.txt" https://storage.googleapis.com/shavar-lists-ios-public/Public/version.txt
+    - name: Locally generate shavar JSON files
+      run: |
+        ls
+        mv sample_shavar_list_creation.ini shavar_list_creation.ini
+
+        echo "Run script to get JSON files"
+        python3 lists2safebrowsing.py
+
+        echo "Lists are generated"
+        ls
+
+        # Once the files are generated, check which is the latest version of these files
+        LATEST_SHAVAR_VERSION=$(ls -1 | grep -v 'gha-creds\-*' | egrep -o '[0-9]{3}.[0-9]'| sort -n | tail -n 1 | sed 's/\.0//g')
+        echo $LATEST_SHAVAR_VERSION
+
+        CURRENT_VERSION=`cat version.txt`
+        echo $CURRENT_VERSION
+        SHOULD_UPDATE_VERSION=false
+
+        # Compare with version saved in the bucket
+        if [ "$LATEST_SHAVAR_VERSION" -gt "$CURRENT_VERSION" ];
+        then
+          echo "Yes, let's get new JSON files"
+          SHOULD_UPDATE_VERSION=true
+          echo "SHOULD_UPDATE_VERSION=$SHOULD_UPDATE_VERSION" >> $GITHUB_ENV
+
+          # Create new folder with the new version
+          mkdir "shavar_version_${LATEST_SHAVAR_VERSION}"
+
+          # Copy all files into a new folder named with the latest version
+          cp -r ${LATEST_SHAVAR_VERSION}.0*.json shavar_version_${LATEST_SHAVAR_VERSION}
+          echo ls shavar_version_${LATEST_SHAVAR_VERSION}
+
+          cd shavar_version_${LATEST_SHAVAR_VERSION}
+
+          # Once the files are stored in their versioned folder, remove the version from the file name
+
+          for file in *; do
+            echo $file
+            #This works, now we need to use variable instead of that value
+            mv "$file" "$(echo $file | sed -E "s/$LATEST_SHAVAR_VERSION\.0-//g")"
+          done
+
+          # Update the version in the version.txt file
+          cd ..
+          sed -i -e "s/$CURRENT_VERSION/$LATEST_SHAVAR_VERSION/g" version.txt
+          UPDATE_SHAVAR_VERSION=`cat version.txt`
+          echo "$UPDATE_SHAVAR_VERSION"
+          echo "UPDATE_SHAVAR_VERSION=$UPDATE_SHAVAR_VERSION" >> $GITHUB_ENV
+        else
+          echo "There is not any new version"
+        fi
+    - name: Get disconnect-entitylist file
+      if: ${{ env.SHOULD_UPDATE_VERSION }}
+      run: |
+        git clone https://github.com/mozilla-services/shavar-prod-lists.git
+        cd shavar-prod-lists && git checkout ${{ env.UPDATE_SHAVAR_VERSION }}.0
```

Comment:
To get the `disconnect-entitylist.json` file, we go the the new versioned branch